tutorial/04_embedding_net.ipynb

```diff
@@ -0,0 +1,385 @@
+      "text/plain": [
+       "<Figure size 802.8x403.92 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# simulate samples\n",
+    "true_parameter = torch.tensor([0.70, np.pi/4])\n",
+    "x_observed, x_points = simulator_model(true_parameter, return_points=True)\n",
+    "\n",
+    "# plot the observation\n",
+    "fig, ax = plt.subplots(facecolor='white', figsize=(11.15, 5.61), ncols=2, constrained_layout=True)\n",
+    "circle = plt.Circle((0, 0), 1.0, color='k', ls='--', lw=0.8, fill=False)\n",
+    "ax[0].add_artist(circle)\n",
+    "ax[0].scatter(x_points[:,0], x_points[:,1], s=20)\n",
+    "ax[0].set_xlabel('x')\n",
+    "ax[0].set_ylabel('y')\n",
+    "ax[0].set_xlim(-1, +1)\n",
+    "ax[0].set_xticks([-1, 0.0, +1.0])\n",
+    "ax[0].set_ylim(-1, +1)\n",
+    "ax[0].set_yticks([-1, 0.0, +1.0])\n",
+    "ax[0].set_title(r'original simulated points with $r = 0.70$ and $\\theta = \\pi/4$')\n",
+    "ax[1].imshow(x_observed.view(32, 32), origin='lower', cmap='gray')\n",
+    "ax[1].set_xticks([]); ax[1].set_yticks([])\n",
+    "ax[1].set_title('noisy observed data (gray image with 32 x 32 pixels)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining an `embedding_net`\n",
+    "\n",
+    "An inference procedure (SNPE, SNLE, SNRE or any other method) applied to the output data from this simulator model determines the posterior distribution of $r$ and $\\theta$ given an observation of $x$, which lives in a 1024 dimensional space (32 x 32 = 1024). To avoid working directly on these high-dimensional vectors, we may use a convolutional neural network (CNN) that takes the 32x32 images as input and returns 8-dimensional feature vectors. This CNN is trained along with the neural density estimator of the inference procedure and serves as an automatic summary statistics extractor. \n",
```

Comment:
Small suggestion:

To avoid working directly on these high-dimensional vectors, one can use a convolutional neural network (CNN) that takes the 32x32 images as input and encodes them into 8-dimensional feature vectors.

Reply:
Also, maybe drop the `(SNPE, SNLE, SNRE or any other method)` because it can be confusing that `SNLE` can not use the `embedding_net`